src/TestSuite/Fixture/FixtureManager.php

```diff
@@ -237,7 +238,7 @@ protected function _setupTable($fixture, $db, array $sources, $drop = true)
         $exists = in_array($table, $sources);
 
         if (($drop && $exists) ||
-            ($exists && !$isFixtureSetup && $fixture->schema() instanceof Table)
+            ($exists && !$isFixtureSetup && $fixture instanceof TableSchemaInterface && $fixture->schema() instanceof Table)
```

Comment:
I'm fine with the interface. I only brought the `method_exists()` option up as the interface doesn't do a lot right now, but if you see potential for future use I'm ok with it.